pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", MonConfigMap, err)
+	}
+
+	cephFsid := string(fsidSecret.Data["fsid"])
+	if cephFsid == "" {
+		return fmt.Errorf("ceph cluster fsid not found")
+	}
+
+	fmt.Println("Check for the running toolbox")
+
+	toolboxDeploymentSpec, err := debug.VerifyDeploymentExists(context, clusterNamespace, "rook-ceph-tools")
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-tools. %v", err)
+	}
+
+	toolBox, err := debug.WaitForPodToRun(context, clusterNamespace, "", toolboxDeploymentSpec.Spec)
+	if err != nil || toolBox == "" {
+		return fmt.Errorf("failed to get the running toolbox")
+	}
+
+	fmt.Printf("Restoring mon quorum to mon %s %s\n", goodMon, goodMonPublicIp)
+	fmt.Printf("The mons to discard are: %s\n", badMons)
+	fmt.Printf("The cluster fsid is %s\n", cephFsid)
+
+	var answer, output string
+	fmt.Printf("Are you sure you want to restore the quorum to mon %s? yes-really-restore\n", goodMon)
+	fmt.Scanf("%s", &answer)
+	output, err = promtToContinueOrCancel(answer)
+	if err != nil {
+		return fmt.Errorf(" restoring the mon quorum to mon %s cancelled", goodMon)
+	}
+	fmt.Println(output)
+
+	fmt.Println("Waiting for operator and mon pods to stop")
+	err = debug.SetDeploymentScale(context, clusterNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		return fmt.Errorf("failed to stop deployment rook-ceph-operator. %v", err)
+	}
+
+	debug.StartDebug(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", goodMon), "")
+
+	debugDeploymentSpec, err := debug.VerifyDeploymentExists(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-mon-%s-debug", goodMon)
+	}
+
+	labelSelector := fmt.Sprintf("ceph_daemon_type=%s,ceph_daemon_id=%s", debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_type"], debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_id"])
+	_, err = debug.WaitForPodToRun(context, clusterNamespace, labelSelector, debugDeploymentSpec.Spec)
+	if err != nil {
+		return fmt.Errorf("failed to start deployment %s", fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	}
+
+	fmt.Println("Started debug pod, restoring the mon quorum in the debug pod")
+
+	monmapPath := "/tmp/monmap"
+
+	fmt.Println("Extracting the monmap")
+	args := []string{
+		fmt.Sprintf("--fsid=%s", cephFsid),
+		"--keyring=/etc/ceph/keyring-store/keyring",
+		"--log-to-stderr=true",
+		"--err-to-stderr=true",
+		"--mon-cluster-log-to-stderr=true",
+		"--log-stderr-prefix=debug",
+		"--default-log-to-file=false",
+		"--default-mon-cluster-log-to-file=false",
+		"--mon-host=$(ROOK_CEPH_MON_HOST)",
+		"--mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS)",
+		fmt.Sprintf("--id=%s", goodMon),
+		"--foreground",
+		fmt.Sprintf("--public-addr=%s", goodMonPublicIp),
+		fmt.Sprintf("--setuser-match-path=/var/lib/ceph/mon/ceph-%s/store.db", goodMon),
+		"--public-bind-addr=$(ROOK_POD_IP)",
```

Comment:
I'm getting here ```Extracting the monmap
parse error setting 'public_bind_addr' to '$(ROOK_POD_IP)'```
@travisn any idea?

Reply:
https://github.com/rook/kubectl-rook-ceph/actions/runs/4465810745/jobs/7843239891?pr=88#step:6:94

Reply:
I see in the test for the restore-quorum function in the script, this is the output:
```
parse error setting 'public_bind_addr' to ''
```

When resetting the monmap the bind address wouldn't matter, so try just setting it to blank.